### PR TITLE
cloud/amazon/aws_config_recorder: add empty resourceTypes list if undefined

### LINKED ISSUE #49453: aws_config_recorder module always reports changed if resource_types list is not specified
##### SUMMARY

aws_config_recorder module will report as Changed if resource_types list attribute is not specified.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

aws_config_recorder module

##### ANSIBLE VERSION

```paste below
ansible 2.7.1
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.12 (default, Nov 12 2018, 14:36:49) [GCC 5.4.0 20160609]
```

##### CONFIGURATION

```paste below
N/A
```

##### OS / ENVIRONMENT

N/A

##### STEPS TO REPRODUCE

Create an AWS Config configuration recorder as follows:

```yaml
- aws_config_recorder:
    name: DefaultConfigRecorder
    role_arn: arn:aws:iam::123412341234:role/ConfigRole
    recording_group:
      all_supported: true
      include_global_types: true
    state: present
```

##### EXPECTED RESULTS

Ansible to report `Changed = true` after the first execution (i.e. the initial configuration), and `Changed = false` after all subsequent executions (i.e. no further changes applied).

##### ACTUAL RESULTS

Ansible reports `Changed = true` every time the task is executed.

### PR DESCRIPTION
**Pull request by slapula**
##### SUMMARY
`recordingGroup[].resourceTypes` exists even if it is not defined as a parameter for a AWS Config recorder.  When the module compares the parameters it has versus what is returned from AWS, the module then determines that there is a change due to the parameter mismatch (and thus will always return `changed`). This PR should fix this issue.  If the parameter is not set when the module is executed, this change adds a condition that will add an empty `resourceTypes` list.

Fixes #49453 

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
`aws_config_recorder`

##### ADDITIONAL INFORMATION
```
N/A
```

### CODE REVIEW

**Comment by ichekaldin** [comment]
@slapula is there anything else that needs to be done before this PR can be merged?

**Comment by slapula** [comment]
@ichekaldin this needs a proper review by the maintainers first then, if there are no issues, can it be merged.

**Comment by Akasurde** [comment]
@ichekaldin Does this PR works for you ? 

needs_info

**Comment by ichekaldin** [comment]
@Akasurde  Yes, I think that does the job. Thank you!

**Review by jillr — APPROVED** [review]

**Pull request merged**
